napi_basics.md

```diff
@@ -0,0 +1,120 @@
+# N-API - Basics
+
+N-API is an API exposed by Node to developers that allow them to author Native add-ons 
+in an ABI stable manner. This means that these add-ons can work against future versions
+of Node without recompilation, and developers get the additional benefit that their 
+add-ons can work independent of the underlying JavaScript Virtual Machine being used 
+by Node.
+
+APIs exposed in the N-API surface are generally used to create and manipulate JavaScript
+objects. Concepts and operations generally map to ideas specified in the ECMA262 Language
+SPecification. The APIs generally have the following properties:
+- All N-API APIs generally return a status code of type `napi_status`. This status 
+can be used to determine whether the API call succeeded or failed.
+- The actual value being returned by the API is returned in an out parameter.
+- All JavaScript values are abstracted behind an opaque type named `napi_value`.
+- In case of an error status code, additional information can be obtained using 
+`napi_get_last_error_info`
+
+## N-API types
+
+N-API exposes the following fundamental datatypes as abstractions that are consumed by 
+the various APIs. These APIs should be treated as opaque types that can be introspected
+only with other N-API APIs.
+
+### *napi_env*
+
+`napi_env` is used to represent a context that the underlying N-API implementation can use 
+to persist VM specific state.
```

Comment:
We can mention that it is passed to native functions when they are invoked and that it must be passed back when making N-API calls.  In addition, the same napi_env that was passed when the initial native function was called should passed into any N-API calls (ie don't cache it once and then use everywhere).